File: src/test/java/no/nav/syfo/config/KafkaTestConfig.java
package no.nav.syfo.config;

import org.junit.ClassRule;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.annotation.EnableKafka;
import org.springframework.kafka.core.DefaultKafkaProducerFactory;
import org.springframework.kafka.core.ProducerFactory;
import org.springframework.kafka.test.rule.KafkaEmbedded;

import static no.nav.syfo.kafka.producer.OversikthendelseProducer.OVERSIKTHENDELSE_TOPIC;
import static org.springframework.kafka.test.utils.KafkaTestUtils.producerProps;

@Configuration
@EnableKafka
public class KafkaTestConfig {

    @ClassRule
    public static KafkaEmbedded embeddedKafka = new KafkaEmbedded(1, true, OVERSIKTHENDELSE_TOPIC);

    @Bean
    public ProducerFactory<String, String> producerFactory() {
        return new DefaultKafkaProducerFactory<>(producerProps(embeddedKafka));
    }
}

File: src/test/kotlin/no/nav/syfo/config/KafkaTestConfig.kt
package no.nav.syfo.config

import no.nav.syfo.oversikthendelse.OversikthendelseProducer
import org.junit.ClassRule
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.annotation.EnableKafka
import org.springframework.kafka.core.DefaultKafkaProducerFactory
import org.springframework.kafka.core.ProducerFactory
import org.springframework.kafka.test.EmbeddedKafkaBroker
import org.springframework.kafka.test.utils.KafkaTestUtils

@Configuration
@EnableKafka
class KafkaTestConfig {
    @Bean
    fun producerFactory(): ProducerFactory<String, String> {
        return DefaultKafkaProducerFactory(KafkaTestUtils.producerProps(embeddedKafka))
    }

    companion object {
        @ClassRule
        var embeddedKafka = EmbeddedKafkaBroker(1, true, OversikthendelseProducer.OVERSIKTHENDELSE_TOPIC)
    }
}
